gravitee-apim-rest-api/gravitee-apim-rest-api-portal/gravitee-apim-rest-api-portal-rest/src/main/java/io/gravitee/rest/api/portal/rest/resource/ApisResource.java

```diff
@@ -189,7 +189,18 @@ protected List<Api> transformPageContent(ExecutionContext executionContext, List
                     api.setLabels(List.of());
                 }
             })
-            .sorted((o1, o2) -> orderingComparator.compare(o1.getId(), o2.getId()))
+            .sorted((o1, o2) -> {
+                if (query != null && !query.isEmpty()) {
+                    // Prioritize APIs containing the query in their names
```

Comment:
You should not force a new sort.
When using a query, you use the SearchEngine Service which already sorts the result.

I think it should be better to avoid applying a new sort in case we have a query.